pallets/dapp-staking-v3/src/types.rs

```diff
@@ -1572,6 +1564,28 @@ impl<NT: Get<u32>> Default for TierParameters<NT> {
     }
 }
 
+/// Extracts the threshold values from a list of `TierThreshold` and calculates
+/// the corresponding balance values for each tier.
+/// The total issuance of the native currency is used to calculate percentage-based thresholds.
+pub fn extract_threshold_values<NT: Get<u32>>(
+    thresholds: BoundedVec<TierThreshold, NT>,
+    total_issuance: Balance,
+) -> BoundedVec<Balance, NT> {
+    thresholds
+        .into_iter()
+        .map(|t| match t {
+            TierThreshold::FixedTvlAmount { amount } => amount,
+            TierThreshold::DynamicTvlAmount { amount, .. } => amount,
+            TierThreshold::FixedPercentage {
+                required_percentage,
+            } => required_percentage * total_issuance,
+            TierThreshold::DynamicPercentage { percentage, .. } => percentage * total_issuance,
+        })
+        .collect::<Vec<Balance>>()
+        .try_into()
+        .expect("Invalid number of tier thresholds provided.")
+}
```

Comment:
Would be more neat to implement a method on the enum that converts it into _threshold_, based on the total issuance.

You could even do it via `From` if you provide a `Get` type that provides the total issuance.